### PR TITLE
fix: `systemjs` re-traverses helpers

### LINKED ISSUE #16219: [Bug]: toPrimitive() does not use the typeof() function
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/cli

### Input code

[Correct](https://babel.dev/repl#?browsers=chrome%2022%2C%20last%2012%20years&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYGwhgzhAECCAeBLA9hAEgUzAEwwJxgG8AoaaAbQGUBPAWwCNkQA6RAF3zDeTwF0AKAJTQSZMngxsArngB2AblLQAvsVVA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.23.8&externalPlugins=&assumptions=%7B%7D)

[Correct - modules false](https://babel.dev/repl#?browsers=chrome%2022%2C%20last%2012%20years&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdgziA2CmB00QHMAUAXAngB1iAZgAQBuAhtAK6yEC8dhARCAEYBWsw6DAlANwBQ_YNFIQIhAIIAPAJYgIACVikAJrABO4gN79ChANoBlTAFtmMeDPQbS6EOoC6qboR16962OgrqwAvQC-_EFAA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.23.8&externalPlugins=&assumptions=%7B%7D)

[Incorrect - modules systemjs](https://babel.dev/repl#?browsers=chrome%2022%2C%20last%2012%20years&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdgziA2CmB00QHMAUAXAngB1iAZgAQBuAhtAK6yEC8dhARCAEYBWsw6DAlANwBQ_YNFIQIhAIIAPAJYgIACVikAJrABO4gN79ChANoBlTAFtmMeDPQbS6EOoC6qboR16962OgrqwAvQC-_EFAA&debug=false&forceAllTransforms=false&modules=systemjs&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.23.8&externalPlugins=&assumptions=%7B%7D)

### Configuration file name

_No response_

### Configuration

```json
{
	"presets": [
		[
			"@babel/preset-env",
			{
				"targets": ["chrome 22", "last 12 years"],
				"modules": "systemjs"
			}
		]
	]
}

```

### Current and expected behavior

For SystemJS, Babel does not replace the typeof operator with the @babel/helpers typeof function in toPrimitive in certain scenarios.

At the same time, in the browser, I get the error '@@toPrimitive must return a primitive value' because the Symbol from the polyfill is treated as an object, not as 'symbol'

### Environment

   "@babel/cli": "7.23.4",
   "@babel/core": "7.23.7",
   "@babel/preset-env": "7.23.8",
    OS: macOS 14.2.1
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    Yarn: 4.0.2 - ~/.nvm/versions/node/v18.15.0/bin/yarn
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm

### Possible solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #16219 
| Patch: Bug Fix?          | √
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | √
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Given that the inline helper is difficult to read and if we regress it's likely we would just think "oh a change in the helper, lets commit it", could you add an `exec.js` test like the one described in the issue?

**Comment by liuxingbaoyu** [comment]
I tried, but we can't use exec.js with systemjs and I haven't figured out a good way to test this.

**Comment by liuxingbaoyu** [comment]
CI is weird and fails randomly.
May be related to https://github.com/zloirock/core-js/issues/1292.

**Review by nicolo-ribaudo — COMMENTED** [review]
I'm debugging this locally

File: packages/babel-plugin-transform-modules-systemjs/test/index.js [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+    );
+    let ret;
+
+    const System = {
+      register: function (_, module) {
+        ret = module().execute();
+      },
+    };
+
+    eval((require, System, content));
```
[the remark this answers is not in the document]
Comment (nicolo-ribaudo) [reply packages/babel-plugin-transform-modules-systemjs/test/index.js]: Ok let's see if this makes the flakyness go away

[the remark this answers is not in the document]
Comment (liuxingbaoyu) [reply packages/babel-plugin-transform-modules-systemjs/test/index.js]: This is just to make eslint happy. :)
I tried using `vm`, but it was blocked by `require`, and I didn't think of `runCodeInTestContext`.

**Comment by liuxingbaoyu** [comment]
@nicolo-ribaudo Thank you!

File: packages/babel-plugin-transform-modules-systemjs/test/fixtures/preset-env/requeue-helpers/input.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Ref: https://github.com/babel/babel/issues/16219
+
+// delete global.Symbol doesn't work with jest in node 6
+Object.defineProperty(global, "Symbol", {
+  configurable: true,
+  writable: true,
+  value: undefined
+});
```
Comment (liuxingbaoyu) [inline packages/babel-plugin-transform-modules-systemjs/test/fixtures/preset-env/requeue-helpers/input.js:8]: To be honest, I don't understand how jest affects our isolation environment in node 6.

Comment (nicolo-ribaudo) [reply packages/babel-plugin-transform-modules-systemjs/test/fixtures/preset-env/requeue-helpers/input.js:8]: Jest does bad things to the global object to implement isolation-but-not-too-much, this workaround is good enough.

**Review by JLHwung — APPROVED** [review]

**Pull request merged**